aws/biganimal-csp-setup

```diff
@@ -239,7 +239,7 @@ create_iam_policy()
s:UpdateAddon","eks:UpdateClusterVersion","eks:UpdateNodegroupConfig","eks:UpdateNodegroupVersion","iam:AttachRolePolicy","iam:CreateOpenIDConnectProvider","iam:CreatePolicy","iam:CreateRole","iam:DeleteOpenIDConnectProvider","iam:DeletePolicy","iam:DeleteRole","iam:DeleteRolePolicy","iam:DetachRolePolicy","iam:GetAccountSummary","iam:GetOpenIDConnectProvider","iam:GetPolicy","iam:GetPolicyVersion","iam:GetRole","iam:GetRolePolicy","iam:ListAccessKeys","iam:ListAttachedRolePolicies","iam:ListEntitiesForPolicy","iam:ListInstanceProfilesForRole","iam:ListInstanceProfileTags","iam:ListMFADevices","iam:ListMFADeviceTags","iam:ListOpenIDConnectProviders","iam:ListOpenIDConnectProviderTags","iam:ListPolicies","iam:ListPoliciesGrantingServiceAccess","iam:ListPolicyTags","iam:ListPolicyVersions","iam:ListRolePolicies","iam:ListRoles","iam:ListRoleTags","iam:ListSAMLProviders","iam:ListSAMLProviderTags","iam:ListServerCertificates","iam:ListServerCertificateTags","iam:ListServiceSpecificCredentials","iam:ListSigningCertificates","iam:ListSSHPublicKeys","iam:ListVirtualMFADevices","iam:PutRolePolicy","iam:TagOpenIDConnectProvider","iam:TagPolicy","iam:TagRole","iam:UntagOpenIDConnectProvider","iam:UpdateAssumeRolePolicy","iam:UpdateRole","kms:CreateAlias","kms:CreateGrant","kms:CreateKey","kms:DeleteAlias","kms:DescribeKey","kms:EnableKeyRotation","kms:GetKeyPolicy","kms:GetKeyRotationStatus","kms:ListAliases","kms:ListResourceTags","kms:ScheduleKeyDeletion","kms:TagResource","kms:UpdateAlias","logs:CreateLogGroup","logs:CreateLogStream","logs:DeleteLogGroup","logs:DeleteMetricFilter","logs:DescribeLogGroups","logs:DescribeMetricFilters","logs:FilterLogEvents","logs:GetLogGroupFields","logs:ListTagsLogGroup","logs:PutLogEvents","logs:PutMetricFilter","logs:PutRetentionPolicy","logs:TagLogGroup","logs:UntagLogGroup","s3:CreateBucket","s3:DeleteBucket","s3:DeleteBucketPolicy","s3:DescribeJob","s3:Get*","s3:List*","s3:PutBucketAcl","s3:PutBucketOwnershipControls","s3:PutBucketPolicy","s3:PutBucketPublicAccessBlock","s3:PutBucketTagging","s3:PutBucketVersioning","s3:PutEncryptionConfiguration","s3:PutObject","servicequotas:ListServiceQuotas","sts:DecodeAuthorizationMessage"],"Effect":"Allow","Resource":"*","Sid":"BigAnimalBasePolicy"},{"Action":["ssm:AddTagsToResource","ssm:GetParameters","ssm:PutParameter","ssm:DeleteParameters"],"Effect":"Allow","Resource":"arn:aws:ssm:*:*:parameter/BA_*","Sid":"BigAnimalSystemsManagerParameterStorePolicy"},{"Action":["iam:PassRole","logs:FilterLogEvents"],"Effect":"Allow","Resource":["arn:aws:iam::*:role/ba-${external_id}-*","arn:aws:iam::*:role/biganimal-${suffix_role_name}","arn:aws:logs:*:*:log-group:/ecs/ba-bootstrap:log-stream:"],"Sid":"BigAnimalECSTaskPolicy"},{"Action":"secretsmanager:*","Effect":"Allow","Resource":"arn:aws:secretsmanager:*:*:secret:BA*","Sid":"BigAnimalSecretsPolicy"},{"Sid":"BigAnimalS3Policy","Effect":"Allow","Action":["s3:DeleteObject","s3:DeleteObjectVersion"],"Resource":"arn:aws:s3:::pg-*/*"}]}
```

Comment:
```suggestion
{
   "Version":"2012-10-17",
   "Statement":[
      {
         "Action":[
            "autoscaling:DescribeAccountLimits",
            "cloudwatch:DeleteDashboards",
            "cloudwatch:GetDashboard",
            "cloudwatch:GetMetricData",
            "cloudwatch:ListMetrics",
            "cloudwatch:PutDashboard",
            "dynamodb:CreateTable",
            "dynamodb:DeleteItem",
            "dynamodb:DescribeContinuousBackups",
            "dynamodb:DescribeTable",
            "dynamodb:GetItem",
            "dynamodb:PutItem",
            "dynamodb:TagResource",
            "dynamodb:UntagResource",
            "ec2:AllocateAddress",
            "ec2:AssociateRouteTable",
            "ec2:AttachInternetGateway",
            "ec2:AuthorizeSecurityGroupEgress",
            "ec2:AuthorizeSecurityGroupIngress",
            "ec2:CreateEgressOnlyInternetGateway",
            "ec2:CreateInternetGateway",
            "ec2:CreateNatGateway",
            "ec2:CreateNetworkAcl",
            "ec2:CreateNetworkAclEntry",
            "ec2:CreateRoute",
            "ec2:CreateRouteTable",
            "ec2:CreateSecurityGroup",
            "ec2:CreateSubnet",
            "ec2:CreateTags",
            "ec2:CreateVpc",
            "ec2:CreateVpcEndpoint",
            "ec2:DeleteEgressOnlyInternetGateway",
            "ec2:DeleteInternetGateway",
            "ec2:DeleteNatGateway",
            "ec2:DeleteNetworkAcl",
            "ec2:DeleteNetworkAclEntry",
            "ec2:DeleteRoute",
            "ec2:DeleteRouteTable",
            "ec2:DeleteSecurityGroup",
            "ec2:DeleteSubnet",
            "ec2:DeleteTags",
            "ec2:DeleteVpc",
            "ec2:DeleteVpcEndpoints",
            "ec2:Describe*",
            "ec2:DetachInternetGateway",
            "ec2:DisassociateRouteTable",
            "ec2:Get*",
            "ec2:ListSnapshotsInRecycleBin",
            "ec2:ModifySubnetAttribute",
            "ec2:ModifyVpcAttribute",
            "ec2:ModifyVpcEndpoint",
            "ec2:ReleaseAddress",
            "ec2:ReplaceNetworkAclAssociation",
            "ec2:ReplaceRoute",
            "ec2:ReplaceRouteTableAssociation",
            "ec2:RevokeSecurityGroupEgress",
            "ec2:RevokeSecurityGroupIngress",
            "ec2:SearchLocalGatewayRoutes",
            "ec2:SearchTransitGatewayRoutes",
            "ecs:CreateCluster",
            "ecs:DeleteCluster",
            "ecs:Describe*",
            "ecs:List*",
            "ecs:RegisterTaskDefinition",
            "ecs:RunTask",
            "ecs:TagResource",
            "eks:CreateAddon",
            "eks:CreateCluster",
            "eks:CreateNodegroup",
            "eks:DeleteAddon",
            "eks:DeleteCluster",
            "eks:DeleteNodegroup",
            "eks:Describe*",
            "eks:List*",
            "eks:TagResource",
            "eks:UntagResource",
            "eks:UpdateAddon",
            "eks:UpdateClusterVersion",
            "eks:UpdateNodegroupConfig",
            "eks:UpdateNodegroupVersion",
            "iam:AttachRolePolicy",
            "iam:CreateOpenIDConnectProvider",
            "iam:CreatePolicy",
            "iam:CreateRole",
            "iam:DeleteOpenIDConnectProvider",
            "iam:DeletePolicy",
            "iam:DeleteRole",
            "iam:DeleteRolePolicy",
            "iam:DetachRolePolicy",
            "iam:GetAccountSummary",
            "iam:GetOpenIDConnectProvider",
            "iam:GetPolicy",
            "iam:GetPolicyVersion",
            "iam:GetRole",
            "iam:GetRolePolicy",
            "iam:ListAccessKeys",
            "iam:ListAttachedRolePolicies",
            "iam:ListEntitiesForPolicy",
            "iam:ListInstanceProfilesForRole",
            "iam:ListInstanceProfileTags",
            "iam:ListMFADevices",
            "iam:ListMFADeviceTags",
            "iam:ListOpenIDConnectProviders",
            "iam:ListOpenIDConnectProviderTags",
            "iam:ListPolicies",
            "iam:ListPoliciesGrantingServiceAccess",
            "iam:ListPolicyTags",
            "iam:ListPolicyVersions",
            "iam:ListRolePolicies",
            "iam:ListRoles",
            "iam:ListRoleTags",
            "iam:ListSAMLProviders",
            "iam:ListSAMLProviderTags",
            "iam:ListServerCertificates",
            "iam:ListServerCertificateTags",
            "iam:ListServiceSpecificCredentials",
            "iam:ListSigningCertificates",
            "iam:ListSSHPublicKeys",
            "iam:ListVirtualMFADevices",
            "iam:PutRolePolicy",
            "iam:TagOpenIDConnectProvider",
            "iam:TagPolicy",
            "iam:TagRole",
            "iam:UntagOpenIDConnectProvider",
            "iam:UpdateAssumeRolePolicy",
            "iam:UpdateRole",
            "kms:CreateAlias",
            "kms:CreateGrant",
            "kms:CreateKey",
            "kms:DeleteAlias",
            "kms:DescribeKey",
            "kms:EnableKeyRotation",
            "kms:GetKeyPolicy",
            "kms:GetKeyRotationStatus",
            "kms:ListAliases",
            "kms:ListResourceTags",
            "kms:ScheduleKeyDeletion",
            "kms:TagResource",
            "kms:UpdateAlias",
            "logs:CreateLogGroup",
            "logs:CreateLogStream",
            "logs:DeleteLogGroup",
            "logs:DeleteMetricFilter",
            "logs:DescribeLogGroups",
            "logs:DescribeMetricFilters",
            "logs:FilterLogEvents",
            "logs:GetLogGroupFields",
            "logs:ListTagsLogGroup",
            "logs:PutLogEvents",
            "logs:PutMetricFilter",
            "logs:PutRetentionPolicy",
            "logs:TagLogGroup",
            "logs:UntagLogGroup",
            "s3:CreateBucket",
            "s3:DeleteBucket",
            "s3:DeleteBucketPolicy",
            "s3:DescribeJob",
            "s3:Get*",
            "s3:List*",
            "s3:PutBucketAcl",
            "s3:PutBucketOwnershipControls",
            "s3:PutBucketPolicy",
            "s3:PutBucketPublicAccessBlock",
            "s3:PutBucketTagging",
            "s3:PutBucketVersioning",
            "s3:PutEncryptionConfiguration",
            "s3:PutObject",
            "servicequotas:ListServiceQuotas",
            "sts:DecodeAuthorizationMessage"
         ],
         "Effect":"Allow",
         "Resource":"*",
         "Sid":"BigAnimalBasePolicy"
      },
      {
         "Action":[
            "ssm:AddTagsToResource",
            "ssm:GetParameters",
            "ssm:PutParameter",
            "ssm:DeleteParameters"
         ],
         "Effect":"Allow",
         "Resource":"arn:aws:ssm:*:*:parameter/BA_*",
         "Sid":"BigAnimalSystemsManagerParameterStorePolicy"
      },
      {
         "Action":[
            "iam:PassRole",
            "logs:FilterLogEvents"
         ],
         "Effect":"Allow",
         "Resource":[
            "arn:aws:iam::*:role/ba-${external_id}-*",
            "arn:aws:iam::*:role/biganimal-${suffix_role_name}",
            "arn:aws:logs:*:*:log-group:/ecs/ba-bootstrap:log-stream:"
         ],
         "Sid":"BigAnimalECSTaskPolicy"
      },
      {
         "Action":"secretsmanager:*",
         "Effect":"Allow",
         "Resource":"arn:aws:secretsmanager:*:*:secret:BA*",
         "Sid":"BigAnimalSecretsPolicy"
      },
      {
         "Sid":"BigAnimalS3Policy",
         "Effect":"Allow",
         "Action":[
            "s3:DeleteObject",
            "s3:DeleteObjectVersion"
         ],
         "Resource":"arn:aws:s3:::pg-*/*"
      }
   ]
}
```

Thanks for the explanation. Then what do you think having formatting this line so we can later knows what is updated from the PR?

Reply:
I agree, having it in a single line compressed JSON is not easy to understand at all (initially probably this was done because of the characters limitation of any IAM object but we can address it better by compressing its content later in the script if needed). But if you don't mind I'd like to keep it as it is for this PR and then address it later in another PR in which we also can address the fact of not having two different JSONs to kept updated and in synch as we were saying. WDYT?

Reply:
got it, thanks.